pkg/apis/core/validation/validation.go

```diff
@@ -7165,3 +7165,24 @@ func ValidatePodAffinityTermSelector(podAffinityTerm core.PodAffinityTerm, allow
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(podAffinityTerm.NamespaceSelector, labelSelectorValidationOptions, fldPath.Child("namespaceSelector"))...)
 	return allErrs
 }
+
+func validatePvNodeAffinity(newPvNodeAffinity, oldPvNodeAffinity *core.VolumeNodeAffinity, fldPath *field.Path) field.ErrorList {
```

Comment:
Unfortunately I think we want to only allow  updates from
- `failure-domain.beta.kubernetes.io/zone`  to `topology.kubernetes.io/zone`
- `failure-domain.beta.kubernetes.io/region` to `topology.kubernetes.io/region`

Generically allowing the labels to change where the label equivalence is not well-defined could introduce unrecoverable race conditions: https://github.com/kubernetes/kubernetes/issues/69528#issuecomment-606972452

Reply:
(maybe allow the beta os/arch labels → GA os/arch labels as well)

Reply:
Okay, let me confirm, we only allow label updates from:
- `failure-domain.beta.kubernetes.io/zone` to `topology.kubernetes.io/zone`
- `failure-domain.beta.kubernetes.io/region` to `topology.kubernetes.io/region`
- `beta.kubernetes.io/arch` to `kubernetes.io/arch`
- `beta.kubernetes.io/os` to `kubernetes.io/os`
- `beta.kubernetes.io/instance-type` to `node.kubernetes.io/instance-type`

Reply:
Yup that looks good to me.

Reply:
thanks, updated